miner/payload_building.go

```diff
@@ -69,26 +69,28 @@ func (args *BuildPayloadArgs) Id() engine.PayloadID {
 // the revenue. Therefore, the empty-block here is always available and full-block
 // will be set/updated afterwards.
 type Payload struct {
-	id           engine.PayloadID
-	empty        *types.Block
-	emptyWitness *stateless.Witness
-	full         *types.Block
-	fullWitness  *stateless.Witness
-	sidecars     []*types.BlobTxSidecar
-	requests     [][]byte
-	fullFees     *big.Int
-	stop         chan struct{}
-	lock         sync.Mutex
-	cond         *sync.Cond
+	id            engine.PayloadID
+	empty         *types.Block
+	emptyWitness  *stateless.Witness
+	full          *types.Block
+	fullWitness   *stateless.Witness
+	sidecars      []*types.BlobTxSidecar
+	emptyRequests [][]byte
```

Comment:
We keep two blocks here, the 'empty' one without transactions, and one with.

The emptyRequests are needed because there has to be an item for each request type even when there are no requests.